alica_engine/include/engine/RunningPlan.h

```diff
@@ -115,17 +116,14 @@ class RunningPlan : public enable_shared_from_this<RunningPlan> {
     bool allChildrenStatus(PlanStatus ps);
     bool anyChildrenTaskSuccess();
     void activate();
-    EntryPoint* getActiveEntryPoint() const { return _activeEntryPoint; }
+    const EntryPoint* getActiveEntryPoint() const { return _activeEntryPoint; }
     void setActiveEntryPoint(EntryPoint* activeEntryPoint);
-    void limitToRobots(std::unordered_set<const supplementary::AgentID*, supplementary::AgentIDHash,
-            supplementary::AgentIDEqualsComparator>
-                    robots);
+    void limitToRobots(const AgentSet& robots);
```

Comment:
AgentIDEqualsComparator can be removed from supplementary since its not used anywhere except for AgentIDManager. AgentIDmanager can also be modified to use AgentSet

Reply:
No, AgentIDEqualsComparator belongs with AgentID. And that package should not depend on alica, it would be circular.